### PR TITLE
Load only the localizations we need, on demand

### PR DESCRIPTION
**Pull request by bhollis**
This refactors our i18next setup to load translations from a separate file, only loading the user's selected language where possible. We've ended up with a lot of translations, so each language is now around 35K. With all 11 possible languages, that was adding a ton to our overall main bundle size (almost half the size!) and adding to memory and parse times.

After this, we load languages on demand from a separate JSON file. The user's preferred language is stored in localStorage for fast lookup, and applied *before* Angular gets bootstrapped, so we don't have to worry about not having translations. The end result should be faster overall loading. I also noticed that development builds sped up a lot.

### CODE REVIEW

**Comment by bhollis** [comment]
For reference, our main.js goes from 843K to 443K with this change.

File: src/app/i18n.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// tslint:disable:no-implicit-dependencies
+import en from 'file-loader?name=[name]-[hash:6].[ext]!../locale/dim.json';
```
Comment (kyleshay) [inline src/app/i18n.ts]: what is this sorcery?? 🔮

File: src/app/settings/dimSettingsService.factory.js [hunk 2]
```diff
@@ -110,12 +103,16 @@ export function SettingsService($rootScope, SyncService, $window, $i18next, $q)
...
 
     $rootScope.$evalAsync(() => {
+      const languageChanged = savedSettings.language !== i18next.language;
       angular.merge(settings, savedSettings);
-      changeLanguage(settings.language, () => {
-        $rootScope.$applyAsync(() => {
-          $rootScope.$broadcast('i18nextLanguageChange');
+      localStorage.dimLanguage = settings.language;
+      if (languageChanged) {
+        changeLanguage(settings.language, () => {
```
Comment (kyleshay) [inline src/app/settings/dimSettingsService.factory.js]: nice, so we still don't need to force a full refresh to load a new language?

Comment (bhollis) [reply src/app/settings/dimSettingsService.factory.js]: Nope, I fixed that when I redid settings. You still need to reload for item translations to update, but we show a handy button for that.

**Pull request merged**